### DISCUSSION TITLE
change WG_HOST env variable without deleting container

### DISCUSSION THREAD

**Discussion by cad0p** [wg-easy/wg-easy#428] [General]
Hi, I have a pretty stable IP address, but I haven't set up DynDNS and after a while my WAN IP changes (my public address).

I saw that it's not possible to change env variables on docker containers while they are running, so do you have other ideas on how to accomplish this without losing my client configurations?

Thanks

**Comment by cad0p** [comment]
Right now my workaround is to delete the entire container, but that is not ideal as you lose the configuration (do you?)

**Reply by eandersons** [reply]
Configuration is not lost if [a Docker volume](https://docs.docker.com/storage/volumes/) is used:

- `-v ~/.wg-easy:/etc/wireguard` if [`docker run`](/WeeJeWel/wg-easy#2-run-wireguard-easy) is used;
- `volumes` section in [`docker-compose.yml`](/WeeJeWel/wg-easy/blob/master/docker-compose.yml) if Docker Compose is used.

**Reply by cad0p** [reply]
Ok thanks, however I think I had a problem that I ran the run command not as sudo, and the folder was created with permissions that do not allow the pi user to see the files in that folder, so that might be why I lost the configuration this time. 

Should I run the command as sudo or change the permissions of that folder?

**Reply by eandersons** [reply]
As far as I know, `sudo` should not affect permissions of a mounted directory.
I am a bit confused about this "pi user" - is this user from the host, WireGuard Easy container or other container?
